engine/src/main/java/org/hibernate/search/engine/metadata/impl/TypeMetadata.java

```diff
@@ -427,13 +428,13 @@ public boolean isJpaIdUsedAsDocumentId() {
 		private final Set<BridgeDefinedField> classBridgeDefinedFields = new HashSet<>();
 
 		public Builder(Class<?> indexedType, ConfigContext configContext) {
-			this( indexedType, new ScopedAnalyzer( configContext.getDefaultAnalyzer() ) );
+			this( indexedType, new ScopedAnalyzerReference( configContext.getDefaultAnalyzer() ) );
 		}
 
-		public Builder(Class<?> indexedType, ScopedAnalyzer scopedAnalyzer) {
+		public Builder(Class<?> indexedType, ScopedAnalyzerReference scopedAnalyzer) {
 			this.indexedType = indexedType;
 			this.scopedAnalyzer = scopedAnalyzer;
-			this.passThroughAnalyzer = PassThroughAnalyzer.INSTANCE;
+			this.passThroughAnalyzer = new LuceneAnalyzerReference( PassThroughAnalyzer.INSTANCE );
```

Comment:
I suppose we could even have a pass-through singleton instance of `LuceneAnalyzerReference`?